app/controllers/spree/admin/reports_controller_decorator.rb

```diff
@@ -181,20 +189,37 @@ def sales_tax
   end
 
   def bulk_coop
-    # -- Prepare date parameters
-    prepare_date_params params
+    if request.format.json?
```

Comment:
> it reduces controller duplication if all actions would need to be copy/pasted to API controller

That's why I'm saying let's move the logic somewhere else and call it from each controller. For me would be easier to follow because we're adding a lot of `if json ... else html` in the same controller.